docs/user/field-selectors.md

```diff
@@ -0,0 +1,135 @@
+| **Boolean**      | `Equals`: Matches if the value is `true` or `false`.<br> `DoubleEquals`: Matches if the value is `true` or `false` (alternative to `Equals`).<br> `NotEquals`: Matches if the value is the opposite of the specified value.<br> `In`: Matches if the value (`true` or `false`) is in the list.<br><i>**Note:** The list can only contain `true` or `false`, so this operator is limited in use.</i><br> `NotIn`: Matches if the value is not in the list.<br> `Exists`: Matches if the field is present.<br> `DoesNotExist`: Matches if the field is not present. | Boolean format (`true`, `false`)                 |
+| **Array**        | `Contains`: Matches if the array contains the specified value.<br> `NotContains`: Matches if the array does not contain the specified value.<br> `In`: Matches if the array overlaps with the specified values.<br> `NotIn`: Matches if the array does not overlap with the specified values.<br> `Exists`: Matches if the field is present.<br> `DoesNotExist`: Matches if the field is not present.<br><br><i>**Note:** Using `Array[Index]` treats the element as the type defined for the array elements (e.g., string, timestamp, number, boolean).</i> | Array element    |
+| **JSON**        | `Equals`: Matches if the value at the specified key path is an exact match to the provided JSON value.<br> `DoubleEquals`: Matches if the value at the specified key path is an exact match to the provided JSON value (alternative to `Equals`).<br> `NotEquals`: Matches if the value at the specified key path does not match the provided JSON value. If the key path does not exist, it is considered a match.<br> `In`: Matches if the value at the specified key path matches at least one JSON value in the provided list.<br> `NotIn`: Matches if the value at the specified key path does not match any of the JSON values in the list. If the key path does not exist, it is considered a match.<br> `Exists`: Matches if the specified key path is present in the JSON field.<br> `DoesNotExist`: Matches if the specified key path is not present in the JSON field.<br> `Contains`: Matches if the value at the specified key path contains the provided JSON value.<br> `NotContains`: Matches if the value at the specified key path does not contain the provided JSON value. | JSON   |
+
+
+## Supported Fields
+
+The following table lists the fields supported for filtering Flight Control objects:
+
+| Field Name                   | Field Type           |
+|------------------------------|----------------------|
+| `metadata.name`              | `String`             |
+| `metadata.owner`             | `String`             |
+| `metadata.labels`            | `StringArray`        |
+| `metadata.annotations`       | `StringArray`        |
+| `metadata.creationTimestamp` | `Timestamp`          |
+| `metadata.deletionTimestamp` | `Timestamp`          |
+| `spec`                       | `JSON`               |
```

Comment:
@fzdarsky, we already expose the Status field through StatusFilter, which is currently supported by the API.

Modifying the Spec or Status fields could also disrupt clients that use the Go client package to query these structures, as well as CLI with oyaml and ojson and manually parse them.

These JSONB fields are indexed in the same way, indexing all JSON structures within

Reply:
> @fzdarsky, we already expose the Status field through StatusFilter, which is currently supported by the API.

That's fine. I didn't say we shouldn't support anything under `.status`, I'm wondering whether we should start with an allow-list containing a small number of fields that we can expand according to user demand, rather than opening up everything forever (as we won't be able to reduce the list later).

> Modifying the Spec or Status fields could also disrupt clients that use the Go client package to query these structures, as well as CLI with oyaml and ojson that manually parse them.

True. But that's not a good analogy because a) the client-go package can hide changes through defaulting, up/down-conversion, etc. and b) it means we don't place the burden on the user but own it ourselves.

Also, remember that a client-go user would be a software developer familiar with handling API versioning. The user of the field selector would be a Linux SysAdmin.